src/Common/Services/EnvironmentDeploymentHandler.cs

```diff
@@ -228,24 +231,28 @@ public async Task Handle(bool interactive, CustomerEnvironment customerEnvironme
                                 if (timeWaited < timeoutToGetSomeMBMessageTask)
                                 {
                                     cancellationTokenMBMessageReceived = new CancellationTokenSource(timeoutToGetSomeMBMessageTask - timeWaited);
+                                    compositeTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationTokenMainTask.Token, cancellationTokenMBMessageReceived.Token);
                                 }

```

Comment:
This case does not account for when a message was received but too long ago.
It should also run the code in the else below, so a minor restructuring would be nice

Reply:
Done :)